docs/poisoning.md

```diff
@@ -0,0 +1,167 @@
+# Poisoning
+
+Updated May 2022
+
+Amory supports a handful of specific poisoning threat models and attacks.  This document will first describe these, providing enough background for newcomers to get up to speed on what these attacks do.  Then, the peculiarities of the poisoning configs will be addressed, including lots of helpful information about Witches' Brew.  Finally, we will describe the poisoning-specific metrics.
+
+
+## Threat Models
+
+There are currently three threat models handled by Armory: dirty-label backdoor, clean-label backdoor, and Witches' Brew (clean-label gradient matching).  In a backdoor attack, an adversary adds a small trigger, or backdoor, to a small portion of the train set in order to gain control of the the model at test time.
+The trigger is usually a small (but not imperceptible) image superposed on the data, and the adversary's goal is to force the model to misclassify test images that have the trigger applied.  Armory includes several trigger images under `utils/triggers/`.
+
+
+In poisoning attacks, the term _source class_ refers to the label of the image(s) that the adversary hopes to misclassify.  In the case of a targeted attack, the _target class_ is the desired misclassification label.  Neither of these terms describes which class gets poisoned; that depends on the threat model.  All of Armory's poisoning scenarios perform targeted attacks.  For simplicity, most Armory scenarios assume a single source class (all images to misclassify are from the same class) and a single target class (all misclassifications are aiming for the same label).  The exception is Witches' Brew, which accepts images from arbitrary source classes that can all have distinct targets.
```

Comment:
> Neither of these terms describes which class gets poisoned; that depends on the threat model.

My understanding is that _source class_ is only meaningful for DLBD attacks since it's not a useful concept for CLBD and WB attacks. Is that still the case for our implementation?